blobfuse/utilities.cpp

```diff
@@ -19,6 +19,87 @@ std::string prepend_mnt_path_string(const std::string path)
     return str_options.tmpPath + "/root" + path;
 }
 
+// cleanup function to clean cached files that are too old
+void gc_cache()
+{
+
+    while(true){
+
+        //if deque is empty, skip
+        if(cleanup.empty())
+        {
+       	    //run it every 1 second
+            usleep(1000);
+            continue;
+        }
+
+        //check if the closed time is old enough to delete
+        if((time(NULL) - cleanup.front().closed_time) > file_cache_timeout_in_seconds)
```

Comment:
i don't think front() is a thread-safe operation

Reply:
There is only a single thread working on the front of the deque. What is there to be concerned of ?

Reply:
will fix this